rfcs/2021-07-20-8288-csv-enrichment.md

```diff
@@ -0,0 +1,202 @@
+# RFC 8288 - 2021-07-20 - CSV Enrichment
+
+Vector needs to allow users to enrich the events flowing through the topology
+using a CSV file. This RFC proposes adding a VRL function that can perform 
+this lookup.
+
+## Scope
+
+The RFC will cover a simple VRL function that will lookup a single row from a
+CSV file. 
+
+The lookup will be a simple sequential scan. The first version of this
+function will not handle any indexing of this data. Above a certain size
+indexing will provide substantial performance benefits, so it is something that
+we should definitely plan for implementing at a future stage. The RFC will 
+briefly discuss this, but will not go into any specific indexing strategies.
+
+## Motivation
+
+Users would like to enrich events flowing through Vector with extra data 
+provided by a CSV file. 
+
+## Doc-level Proposal
+
+### Resources
+
+To represent the CSV file we have a new top level configuration option.
+
+```
+[resources.csv_file]
```

Comment:
We don't have to outline all of the metrics here, but I think it'd be useful to think about instrumenting these resources so that users can get an idea of things like file i/o.

Reply:
Not sure I fully understand. Do you mean emit metrics for filesize, load time etc?

Reply:
Ah, true, I forgot we were loading into memory at start so there wouldn't really be on-going metrics for it outside of, say, how long a given `lookup_csv_row` call takes which would be a separate concern (but probably interesting to users).